addons/mail/static/src/messaging/utils.js

```diff
@@ -0,0 +1,117 @@
+odoo.define('mail.messaging.utils', function (require) {
+'use strict';
+
+const {
+    patch: webUtilsPatch,
+    unpatch: webUtilsUnpatch,
+} = require('web.utils');
+
+//------------------------------------------------------------------------------
+// Public
+//------------------------------------------------------------------------------
+
+const classPatchMap = new WeakMap();
+
+/**
+ * Inspired by web.utils:patch utility function
```

Comment:
what is the reason not using it instead?

Reply:
Because it only works on instance methods, not class methods.